### ISSUE TITLE
pep.xml export

### ISSUE THREAD

**Issue by wolski** [compomics/peptide-shaker#221]
Just trying to read the pep.xml exported with Peptide shaker 1.13.6. with bioconductor pepXMLTab.

Error: 1: Opening and ending tag mismatch: specificity line 9 and sample_enzyme
2: Opening and ending tag mismatch: aminoacid_modification line 17 and search_summary
3: Opening and ending tag mismatch: mod_aminoacid_mass line 24 and modification_info
4: Opening and ending tag mismatch: parameter line 35 and analysis_result
5: Opening and ending tag mismatch: parameter line 34 and search_hit
6: Opening and ending tag mismatch: mod_aminoacid_mass line 40 and modification_info
7: Opening and ending tag mismatch: parameter line 49 and analysis_result
8: Opening and ending tag mismatch: parameter line 48 and search_hit
9: Opening and ending tag mismatch: mod_aminoacid_mass line 55 and modification_info
10: Opening and ending tag mismatch: parameter line 65 and analysis_result
11: Opening and ending tag mismatch: parameter line 64 and search_hit
12: Opening and ending tag mismatch: mod_aminoacid_mass line 72 and modification_info
13: Opening and ending tag mismatch: parameter line 83 and an

**Comment by mvaudel** [comment]
Hi Vitold,

As far as I can see the tags you mention are properly closed in cour code. This is what I get from the example dataset:

```
		<sample_enzyme name="Trypsin" fidelity="specific">
			<specificity cut="RK" no_cut="P" sense="C">
		</sample_enzyme>
		<search_summary precursor_mass_type="monoisotopic" fragment_mass_type="monoisotopic">
			<aminoacid_modification aminoacid="C" massdiff="57.02146372057" mass="160.03064850528" variable="N" symbol="Carbamidomethylation of C" description="UNIMOD:4">
			<aminoacid_modification aminoacid="M" massdiff="15.99491461956" mass="147.03539953255003" variable="Y" symbol="Oxidation of M" description="UNIMOD:35">
			<terminal_modification terminus="n" massdiff="42.0105646837" protein_terminus="n" mass="43.01838971577" variable="Y" symbol="Acetylation of protein N-term" description="UNIMOD:1">
			<aminoacid_modification aminoacid="E" massdiff="-18.010564683699997" mass="111.03202840427004" peptide_terminus="n" variable="Y" symbol="Pyrolidone from E" description="UNIMOD:27">
			<aminoacid_modification aminoacid="Q" massdiff="-17.026549101009998" mass="111.03202840427002" peptide_terminus="n" variable="Y" symbol="Pyrolidone from Q" description="UNIMOD:28">
			<aminoacid_modification aminoacid="C" massdiff="-17.026549101009998" mass="85.98263568370001" peptide_terminus="n" variable="Y" symbol="Pyrolidone from carbamidomethylated C" description="UNIMOD:385">
		</search_summary>
```

This is what Comet returns for a similar search:
```
 <sample_enzyme name="Trypsin">
  <specificity cut="RK" no_cut="P" sense="C"/>
 </sample_enzyme>
```
```
  <aminoacid_modification aminoacid="M" massdiff="15.994915" mass="147.035400" variable="Y" symbol="*"/>
  <aminoacid_modification aminoacid="C" massdiff="57.021464" mass="160.030649" variable="N"/>
```
Looks the same to me?

That said, PepXML is not a standard format, it is mainly made for the TPP, and we only provide a beta implementation with no guarantee that we managed to do things as the TPP does. In general we recommend using mzIdentML instead, it allows capturing more information, it is better documented, and actively maintained. If you want to chain different tools outside the TPP you might want to stay away from PepXML.

Best regards,

Marc

**Comment by wolski** [comment]
Hi Marc, Please you please note that all the tags in your example ar NOT properly closed. 
The slash / is missing in all of them i.e:

description="UNIMOD:1">
while AFAIK
description="UNIMOD:1"/>

To close them.

**Comment by mvaudel** [comment]
good point, somehow I was sure they were there but you are obviously correct. Thanks for the pointer!

**Comment by mvaudel** [comment]
Hi again,

It seems that the slash was systematically missing. Hopefully it is corrected now. We will release new versions soon containing that fix. Can you try that the file at the link below solves the problem?

Best regards,

Marc

http://vedlegg.uib.no/?id=9f22310b23a5210464897fee4afb5309

**Comment by mvaudel** [comment]
Hi again,

Great, I will check the issue with bibliospec and follow-up on the respective issue.

If you agree I will keep the issue open until we make the new release in case other people get the same problem.

Thanks again for your help,

Marc

**Comment by hbarsnes** [comment]
Hi,

We've now released PeptideShaker v1.14.0 which should fix the pepXML closing tag issue. If this is not the case, please let us know and we'll reopen the issue.

Best regards,
Harald
